c2corg_api/views/document.py

```diff
@@ -493,6 +506,53 @@ def _check_versions(self, document, document_in):
                 raise HTTPConflict('version of geometry has changed')
 
 
+def validate_geometry_type(document):
+    """ Checks that geometry field contains valid geometry for each doc. type:
+        articles, books - no geometry
+        images, xreports, user_profiles, waypoints - should be only Point
+        areas, maps - should be only Polygon
+        outings, routes - should be only LineString or MultiLineString
+        If there is no geometry, nothing has to be validated.
+    :param document:
+    :return:
+    """
+    doc_type = document.type
+
+    if doc_type in [OUTING_TYPE, ROUTE_TYPE, AREA_TYPE, MAP_TYPE]:
+        g_field = str(document.geometry.geom_detail)
+    elif doc_type in [IMAGE_TYPE, USERPROFILE_TYPE,
+                      XREPORT_TYPE, WAYPOINT_TYPE]:
+        g_field = str(document.geometry.geom)
+
+    if g_field == 'None':
+        return True
+
+    try:
+        s = wkb.loads(g_field, hex=True)
+        geom_type = s.type
+    except Exception as e:
+        logging.exception(e)
+        try:
+            q = "SELECT GeometryType(ST_GeomFromEWKT('" + g_field + "'));"
+            result = DBSession.execute(q)
+            geom_type = result.fetchone()[0]
+        except:
+            log.warning('Error with geometry %s %s',
+                        doc_type, g_field, exc_info=True)
+            return False
+
+    if doc_type in [IMAGE_TYPE, USERPROFILE_TYPE, XREPORT_TYPE, WAYPOINT_TYPE]:
+        if geom_type.upper() in ['POINT']:
+            return True
```

Comment:
A shorter syntax would be for instance:
```
geom_type = geom_type.upper()
if doc_type in [IMAGE_TYPE, USERPROFILE_TYPE, XREPORT_TYPE, WAYPOINT_TYPE]:
  return geom_type == 'POINT'
elif doc_type in [OUTING_TYPE, ROUTE_TYPE]:
  return geom_type in ['LINESTRING', 'MULTILINESTRING']
elif doc_type in [AREA_TYPE, MAP_TYPE]:
  return geom_type == 'POLYGON'
return False
```

By the way do you really need to check for ``MAP_TYPE``? In UI they are not even a document type and it's not possible to edit them. Is it possible in the API?